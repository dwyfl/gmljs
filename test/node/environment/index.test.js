import GML from '../../../src';
import GMLEnvironment from '../../../src/node/environment';
import testXml from './data/test.xml';

it('creates a correct default GMLEnvironment node', () => {
  const gml = GMLEnvironment.create().toString();
  expect(gml).toMatchSnapshot();
});

it('creates a correct GMLEnvironment node from spec XML', () => {
  const gmlNode = GML.createNodeFromXml(GMLEnvironment, testXml);
  const gml = gmlNode.toString();
  expect(gml).toMatchSnapshot();
});