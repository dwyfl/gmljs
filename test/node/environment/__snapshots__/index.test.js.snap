// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a correct GMLEnvironment node from spec XML 1`] = `"<environment><offset><x>0</x><y>0</y><z>0</z></offset><rotation><x>0</x><y>0</y><z>0</z></rotation><up><x>0</x><y>-1</y><z>0</z></up><screenbounds><x>1024</x><y>768</y><z>0</z></screenbounds><origin><x>0</x><y>0</y><z>0</z></origin><realscale><x>1000</x><y>600</y><z>0</z><unit>cm</unit></realscale><audio>youraudio.mp3</audio><background>yourimage.jpg</background></environment>"`;

exports[`creates a correct default GMLEnvironment node 1`] = `"<environment><up><x>0</x><y>-1</y><z>0</z></up><screenbounds><x>1920</x><y>1080</y><z>0</z></screenbounds></environment>"`;
